velox/common/memory/MemoryPool.cpp

```diff
@@ -453,18 +452,17 @@ void* MemoryPoolImpl::allocateZeroFilled(int64_t numEntries, int64_t sizeEach) {
         sizeEach,
         toString()));
   }
+  CHECK_AND_INC_MEM_OP_STATS(Allocs);
   return buffer;
 }
 
 void* MemoryPoolImpl::reallocate(void* p, int64_t size, int64_t newSize) {
```

Comment:
Can you update DataBuffer::reserve() to try catch reallocate() failure and free the old buffer in catch?